RUFAS/routines/animal/ration/ration_driver.py

```diff
@@ -88,12 +88,26 @@ def formulate_ration(
         )
         # Reduction of milk production estimate process to achieve feasible solution
         num_reattempts: int = 0
+        info_map = {
+            "class": "RationManager",
+            "function": cls.formulate_ration.__name__,
+        }
 
         # TODO: Put AnimalCombination enum in a separate file and use it here instead of hardcoding the names
         # GitHub Issue #793
         if pen.animal_combination.name in ["LAC_COW"]:
             while not solution.success:
                 num_reattempts += 1
+                if pen.avg_milk < AnimalModuleConstants.MINIMUM_AVG_PEN_MILK:
+                    om.add_error(
+                        "Ration formulation error",
+                        "Critical ration formulation error: milk production has dropped too low."
+                        + f" Check failed_constraint_summary_for_pen_{pen.id} to see what caused formulation to fail."
+                        + " Possible solution is to provide additional feed ingredients to"
+                        + f" {pen.animal_combination.name}.",
+                        info_map,
+                    )
```

Comment:
Not a big deal for this PR as we are not going to hit this string multiple times a run; but:
1. Generally speaking, it is better to break an f-string across multiple lines like this rather using `+` to concatenate.
2. Also, it is better to leave the white space at the end of the last line instead of the beginning of the next line.
3. The name `"Ration formulation error"` is redundant as we already know the class and function names and we know this is an error (we are adding an error after all). We can increase the information density by shuffling the words like this.
```suggestion
                    om.add_error(
                        "milk production too low",
                        (
                           f"Check failed_constraint_summary_for_pen_{pen.id} to see what caused formulation to fail. "
                           f"Possible solution is to provide additional feed ingredients to {pen.animal_combination.name}."
                         ),
                        info_map,
                    )
```

Reply:
Great suggestion, point well taken! Updated accordingly.